app/models/user.rb

```diff
@@ -2,10 +2,14 @@ class User < ApplicationRecord
   has_secure_password
 
   validates :email, uniqueness: true, presence: true
-  validates_presence_of :password, :role, :active, :address, :city, :state, :zip, require: true
-  validates_numericality_of :role, only_integer: true
+  validates_presence_of :name, :role, :active, :address, :city, :state, :zip, require: true
+  validates :password, :presence =>true, :confirmation =>true
+  validates_confirmation_of :password
+  validates :role, numericality: {only_integer: true}
```

Comment:
```suggestion
  validates :role, numericality: {only_integer: true}
```
We can delete the only integer validation because when we use `enum role: %w(user admin merchant)`, it reassigns the value to a string.